keel-core/src/test/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusServiceTests.kt

```diff
@@ -192,13 +192,43 @@ class ResourceStatusServiceTests : JUnit5Minutests {
           }
         }
 
-        context("when last event is ResourceActuationVetoed") {
+        context("when last event is ResourceActuationVetoed because unhappy") {
           before {
-            repository.appendResourceHistory(ResourceActuationVetoed(resource, "vetoed", clock))
+            repository.appendResourceHistory(ResourceActuationVetoed(resource, "vetoed", "UnhappyVeto", null, clock))
           }
 
-          test("returns VETOED") {
-            expectThat(subject.getStatus(resource.id)).isEqualTo(VETOED)
+          test("returns UNHAPPY") {
+            expectThat(subject.getStatus(resource.id)).isEqualTo(UNHAPPY)
+          }
+        }
+
+        context("when last event is ResourceActuationVetoed because new veto that doesn't provide a status") {
```

Comment:
This test seems redundant with the one above.

Reply:
Sure, combining the tests.